examples/elasticsearch/elasticsearch_discovery.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+)
+
+var (
+	username   = flag.String("username", "admin", "Username for basic authentication to the API server.")
+	password   = flag.String("password", "", "Password for basic authentication to the API server.")
+	server     = flag.String("server", "", "The address and port of the Kubernetes API server")
+	namespace  = flag.String("namespace", api.NamespaceDefault, "The namespace containing Elasticsearch pods")
+	labelKey   = flag.String("label_key", "name", "The label key to be used to filter Elasticsearch pods")
```

Comment:
Consider taking a selector as a string-- that way it's easy for user to specify multiple label/key pairs?